src/main/java/org/curriki/api/enus/vertx/WorkerVerticle.java

```diff
@@ -187,69 +225,149 @@ private void importTimer(String classSimpleName) {
 				Duration duration = TimeTool.parseNextDuration(importPeriod);
 				ZonedDateTime nextStartTime = startDateTime.plus(duration);
-				LOG.info(String.format(importTimerScheduling, nextStartTime.format(TIME_FORMAT)));
+				LOG.info(String.format(importTimerScheduling, classSimpleName, nextStartTime.format(TIME_FORMAT)));
 				Duration nextStartDuration = Duration.between(Instant.now(), nextStartTime);
-				vertx.setTimer(nextStartDuration.toMillis(), b -> {
+				vertx.setTimer(nextStartDuration.toMillis(), c -> {
 					importData(classSimpleName, nextStartTime);
 				});
-			});
+			}
 		}
 	}
 
+	private Future<Void> importData() {
+		Promise<Void> promise = Promise.promise();
+		importTimer("CurrikiResource");
+		return promise.future();
+	}
+
 	/**	
-	 * Import initial data
-	 * Val.Complete.enUS:Importing initial data completed. 
-	 * Val.Fail.enUS:Importing initial data failed. 
-	 * Val.Skip.enUS:Skip importing data. 
+	 * Val.Complete.enUS:Importing all CurrikiResource records completed. 
+	 * Val.Fail.enUS:Importing CurrikiResource records failed. 
+	 * Val.Skip.enUS:Skip importing CurrikiResource records. 
 	 **/
-	private Future<Void> importData() {
+	private Future<Void> importDataCurrikiResource(SQLConnection sqlConnection) {
 		Promise<Void> promise = Promise.promise();
-		importTimer("ChoiceDonor");
-		importTimer("ChoiceImage");
+
+		try {
+			sqlConnection.queryStreamWithParams(
+					"SELECT * from currikidb.resources"
+					, new JsonArray(), a -> {
+				SQLRowStream sqlRowStream = a.result();
+				Integer fetchSize = config().getInteger(ConfigKeys.MOONSHOTS_FETCH_SIZE);
+				ApiCounter counter = new ApiCounter();
+				sqlRowStream.pause();
+				sqlRowStream.fetch(fetchSize);
+				sqlRowStream.resultSetClosedHandler(b -> {
+					sqlRowStream.moreResults();
+				}).handler(row -> {
+					counter.incrementQueueNum();
+					processRowCurrikiResource(row).onSuccess(b -> {
+						counter.decrementQueueNum();
+						counter.incrementTotalNum();
+						if(counter.getQueueNum().compareTo(0L) == 0) {
+							sqlRowStream.fetch(fetchSize);
+						}
+					}).onFailure(ex -> {
+						LOG.error(importDataCurrikiResourceFail, ex);
+						promise.fail(ex);
+					});
+				}).exceptionHandler(ex -> {
+					LOG.error(importDataCurrikiResourceComplete, ex);
+					promise.fail(ex);
+				}).endHandler(b -> {
+					LOG.info(importDataCurrikiResourceComplete);
+					promise.complete();
+				});
+			});
+		} catch(Exception ex) {
+			LOG.error(importDataCurrikiResourceFail, ex);
+			promise.fail(ex);
+		}
+
 		return promise.future();
 	}
 
-	private Future<Void> importDataCurrikiResource() {
+	/**	
+	 * Val.Complete.enUS:Importing CurrikiResource row completed. 
+	 * Val.Fail.enUS:Importing CurrikiResource row failed. 
+	 **/
+	private Future<Void> processRowCurrikiResource(JsonArray row) {
 		Promise<Void> promise = Promise.promise();
+
+		try {
+//			String resourceId = row.getString(0);
```

Comment:
do we need this chunk of commented code?

Reply:
Good call @rohit-taneja-redhat , that is where we will integrate the new CurrikiResource class in the import that @dheerajodha is working on.